depends/funcs.mk

```diff
@@ -30,6 +30,23 @@ define fetch_file
     rm -rf $$($(1)_download_dir) ))
 endef
 
+define generate_crate_checksum
+rm .gitignore && \
+echo "{\"files\":{`find . -type f | grep -v "$($(1)_file_name)" | grep -v .stamp_ | sed 's|^./||' | sort | xargs $(build_SHA256SUM) | sed 's|^\([^ ]\+\)  \(.\+\)$$$$|"\2":"\1"|g' | tr '\n' ',' | sed 's|,$$$$||'`},\"package\":\"$($(1)_sha256_hash)\"}" > .cargo-checksum.json
+endef
+
+define vendor_crate_source
+mkdir -p $($(1)_staging_prefix_dir)/$(CRATE_REGISTRY) && \
+cp -r $($(1)_extract_dir) $($(1)_staging_prefix_dir)/$(CRATE_REGISTRY)/$($(1)_crate_name) && \
+cd $($(1)_staging_prefix_dir)/$(CRATE_REGISTRY)/$($(1)_crate_versioned_name) && \
+rm -r `basename $($(1)_patch_dir)` .stamp_* .$($(1)_file_name).hash
+endef
+
+define configure_crate_registry
+mkdir .cargo && \
+echo "[source.vendored-sources]\ndirectory = \"$(host_prefix)/$(CRATE_REGISTRY)\"\n\n[source.crates-io]\nreplace-with = \"vendored-sources\"" > .cargo/config
```

Comment:
These newlines aren't going to work right depending on the platform. (In the homu try, you can see the mac build failed because these weren't escaped.)

Reply:
I think it would be better to have a template file that gets sed'd, allowing us to easily change other configurations or target features.